lib/usd/translators/mayaReferenceUpdater.cpp

```diff
@@ -25,46 +25,86 @@
 #include <pxr/base/gf/vec2f.h>
 #include <pxr/base/tf/diagnostic.h>
 #include <pxr/pxr.h>
+#include <pxr/usd/pcp/layerStack.h>
+#include <pxr/usd/pcp/layerStackIdentifier.h>
+#include <pxr/usd/pcp/node.h>
+#include <pxr/usd/sdf/copyUtils.h>
 #include <pxr/usd/sdf/path.h>
+#include <pxr/usd/usd/payloads.h>
+#include <pxr/usd/usd/primCompositionQuery.h>
 #include <pxr/usd/usd/timeCode.h>
+#include <pxr/usd/usd/variantSets.h>
 #include <pxr/usd/usdGeom/tokens.h>
 #include <pxr/usd/usdUtils/pipeline.h>
 
 PXR_NAMESPACE_OPEN_SCOPE
 
 PXRUSDMAYA_REGISTER_UPDATER(
     MayaUsd_SchemasMayaReference,
+    reference,
     PxrUsdTranslators_MayaReferenceUpdater,
-    (UsdMayaPrimUpdater::Supports::Push | UsdMayaPrimUpdater::Supports::Clear));
+    (UsdMayaPrimUpdater::Supports::Push | UsdMayaPrimUpdater::Supports::Clear
+     | UsdMayaPrimUpdater::Supports::AutoPull));
 PXRUSDMAYA_REGISTER_UPDATER(
     MayaUsd_SchemasALMayaReference,
+    reference,
     PxrUsdTranslators_MayaReferenceUpdater,
-    (UsdMayaPrimUpdater::Supports::Push | UsdMayaPrimUpdater::Supports::Clear));
+    (UsdMayaPrimUpdater::Supports::Push | UsdMayaPrimUpdater::Supports::Clear
+     | UsdMayaPrimUpdater::Supports::AutoPull));
 
 PxrUsdTranslators_MayaReferenceUpdater::PxrUsdTranslators_MayaReferenceUpdater(
     const MFnDependencyNode& depNodeFn,
-    const SdfPath&           usdPath)
-    : UsdMayaPrimUpdater(depNodeFn, usdPath)
+    const Ufe::Path&         path)
+    : UsdMayaPrimUpdater(depNodeFn, path)
 {
 }
 
 /* virtual */
-bool PxrUsdTranslators_MayaReferenceUpdater::Pull(UsdMayaPrimUpdaterContext* context)
+bool PxrUsdTranslators_MayaReferenceUpdater::pushCopySpecs(
+    SdfLayerRefPtr srcLayer,
+    const SdfPath& srcSdfPath,
+    SdfLayerRefPtr dstLayer,
+    const SdfPath& dstSdfPath)
 {
-    const UsdPrim& usdPrim = GetUsdPrim<MayaUsd_SchemasMayaReference>(*context);
-    const MObject& parentNode = GetMayaObject();
+    bool success = false;
 
-    UsdMayaTranslatorMayaReference::update(usdPrim, parentNode);
+    // We are looking for a very specific configuration in here
```

Comment:
Worth mentioning that this code is going to change very soon as it's only a "prototype" code.